Create most of the regions

ChaoticYosher opened the pull request:
Regions.py
- Renamed the Beginning Area Regions
- Added Mailbox HP to the Mailbox Region
- Added paths from Termina Field to Clock Town, Milk Road, and the four regions.
- Added regions for Deku Palace, and the Mountain, Ocean, and Canyon areas, without the connections in place

Rules.py: Added a rule to restrict backwards access to the tutorial area until we make that accessible.

EntranceShuffle.py: Added links for Termina Field, and West/North Clock Town

ChaoticYosher committed:
Create most of the regions

Regions.py
- Renamed the Beginning Area Regions
- Added Mailbox HP to the Mailbox Region
- Added paths from Termina Field to Clock Town, Milk Road, and the four regions.
- Added regions for Deku Palace, and the Mountain, Ocean, and Canyon areas, without the connections in place

Rules.py: Added a rule to restrict backwards access to the tutorial area until we make that accessible.

EntranceShuffle.py: Added links for Termina Field, and West/North Clock Town

Vlix commented on EntranceShuffle.py:
```diff
@@ -91,20 +91,24 @@ def connect_fairy(world, entrancename, exitname):
+mandatory_connections = [ ('Cursed Underground', 'Clock Tower'),
+                                                ('South Mailbox', 'Mailbox'),
                                                 ('East Mailbox', 'Mailbox'),
                                                 ('North Mailbox', 'Mailbox'),
```
Aren't these `South/East/North` mailboxes superfluous? Can't it just be `South/East/North Clock Town` that all connect to `Mailbox`?

Vlix commented on EntranceShuffle.py:
```diff
@@ -122,18 +126,50 @@ def connect_fairy(world, entrancename, exitname):
+                                        ('Trading Post', 'Trading Post'),
+                                        ('Bomb Shop', 'Bomb Shop'),
+                                        ('Post Office', 'Post Office'),
+                                        ('Lottery Shop', 'Lottery Shop'),
```
This is slightly confusing. Maybe keep to the convention of naming the first `Entrance` or `Exit` for clarity?

Vlix commented on Regions.py:
```diff
@@ -5,10 +5,11 @@
 
     world.regions = [
+        create_ow_region('Beginning',[], ['Cursed Underground']),
+        create_interior_region('Deku Flower Tutorial Area', ['First Nuts'], []),
```
Why is the third argument an empty array here?

Vlix commented on Regions.py:
```diff
@@ -73,16 +74,27 @@ def create_regions(world):
         create_interior_region('Swamp Tourist Centre', ['Swamp Tourist Free Product', 'Pictograph Contest Winner'], ['Boat Ride', 'Southern Swamp']),
         create_ow_region('Boat Ride', None, ['Poison Swamp']),
+        create_ow_region('Poison Swamp', [], ['Deku Palace', 'Swamp Spider House Entrance']),
+        create_interior_region('Swamp Spider House'),
```
This doesn't have a way out?

Vlix commented on Regions.py:
```diff
@@ -73,16 +74,27 @@ def create_regions(world):
+        create_interior_region('Deku Palace Shrine'),
+        create_interior_region('Woodfall',
+            [],
+            []),
```
Is this a `TODO:`?

Vlix reviewed (commented):
Just added some comments.
Pro-tip, when adding/changing something and you know you (or someone else) will have to revisit it, because it's tentative or just half of the work, add comments describing what needs to still be done prefixed with `TODO:`. That way it's easy to see if something still needs work, and it's easier for reviewers to know that you also know this isn't finished or correct (yet)

epicYoshi64 commented:
Okay, I'll add TODO comments where I would otherwise have answered your individual review comments, and then resubmit this pull request ( does making commits automatically update the pull request? )

Vlix commented:
Yeah, pushing commits to your remote branch automatically updates the pull
request.

On Fri, Sep 14, 2018, 20:27 Alex Trush <notifications@github.com> wrote:

> Okay, I'll add TODO comments where I would otherwise have answered your
> individual review comments, and then resubmit this pull request ( does
> making commits automatically update the pull request? )
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/epicYoshi64/MM-Randomizer/pull/4#issuecomment-421445252>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AGhhfYyCdcGoISTOYtZhZbMa_JAohFCCks5ua_ULgaJpZM4WmhPm>
> .
>

ChaoticYosher committed:
Location reordering, addition of songs, owl statues, and boss remains, connections up to Mountain Village

Regions.py changes to location_table
- Regrouped entries in location_table by type.
- Introduced a Collectable category for mainly heart pieces, or anything else on the overworld directly.
- Added the songs, boss defeat events, boss heart, and boss remains, and Deku trading quest, and owl statues

Regions.py
- Added the path up to and including Mountain Village
- Made beginning area point to clock tower and removed the chest in that area from the logic until we figure out how to handle it
- South Clock Town: removed the Deku Salesman, and renamed the 50 Rupee chest to Festival Tower Chest
- Clock Tower: made it point to the backwards twisted hallway
- Moved 'Dropped Ocarina', and 'Song From Skull Kid' to Clock Tower directly as to imply we will hand out a song, and one random item at the beginning of the playthrough ... somehow
- Stray Fairy pickup locations removed for now

Entrance Shuffle.py: added links up to Mountain Village

Items.py
- Added Songs
- Added Stray Fairy Pickups
- Added Boss Remains

BaseClasses.py
- Changed open_forest, and open_door toopen_ocarina and shuffle_owls

Notes folder
- Added a note about lullaby intro flag to engine notes
- Added a list of potential gui options
- Added a todo list file
- Added a file for notes about resettable items

Vlix reviewed (approved):
I'm gonna just approve this pull request, seeing as it's just a snapshot of work still in progress. Any next pull request would ideally be a contained, and finished, piece of code.
(Also @RevelationOrange, for reference)
